src/components/lightDarkTheme/LightDarkTheme.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export const LightDarkThemeToggle = () => {
+  const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)')?.matches ? 'dark' : 'light';
+  const [theme, setTheme] = useState(localStorage.getItem('theme') || defaultTheme);
+
+  const toggleTheme = useCallback(() => {
+    const newTheme = theme === 'light' ? 'dark' : 'light';
+    setTheme(newTheme);
+    localStorage.setItem('theme', newTheme);
+  }, [theme]);
+
+  useEffect(() => {
+    applyThemeStyles();
+  }, [theme]);
+
+  const applyThemeStyles = () => {
+    const body = document.body;
+    if (theme === 'light') {
+      body.setAttribute('data-theme', 'light');
+    } else {
+      body.setAttribute('data-theme', 'dark');
+    }
+  };
```

Comment:
All this logic should be excluded in ./hook.ts file for this component.

Reply:
Done